thrust/testing/transform.cu

```diff
@@ -7,8 +7,14 @@
 #include <thrust/iterator/zip_iterator.h>
 #include <thrust/iterator/retag.h>
 
+#if (THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_GCC) && __GNUC__ >= 12 && THRUST_CPP_DIALECT >= 2020
```

Comment:
suggestion: I'd add a comment explaining why this is needed. Same for functional.cu.